Samples for Extending Pentaho/Reference Implementations/Security/SAML 2.0/pentaho-saml/src/main/java/org/pentaho/platform/spring/security/saml/PentahoSamlUserDetailsService.java

```diff
@@ -25,13 +27,14 @@
 
   private static final String PROVIDER_NAME = "providerName";
 
-  UserDetailsService userDetailsService;
-  String selectedAuthorizationProvider;
+  private UserDetailsService userDetailsService;
+  private String selectedAuthorizationProvider;
 
   // if the selectedAuthorizationProvider is our own ( i.e. samlId ), we'll directly use our own
-  UserDetailsService samlUserDetailsService;
-  String samlId;
-
+  private PentahoSamlNativeUserDetailsService samlUserDetailsService;
+  private String samlId;
+  private boolean createDetailsOnUsernameNotFoundException = false;
+  
   public PentahoSamlUserDetailsService( UserDetailsService userDetailsService ){
     setUserDetailsService( userDetailsService );
     Assert.notNull( userDetailsService );
```

Comment:
while we're on it, let's change the order of these 2 lines ( i.e. have the assertion made prior to the setter )